packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -23,6 +24,63 @@ export interface IService extends IResource {
   readonly serviceArn: string;
 }
 
+/**
+ * Properties for ECS container targets.
+ */
+export interface TargetProps {
+  /**
+   * Name of the container.
+   */
+  readonly containerName: string,
+
+  /**
+   * Container targets of the container.
+   */
+  readonly containerTargets: ContainerTargetProps[]
+}
+
+/**
+ * Properties for registering a container target.
+ */
+export interface ContainerTargetProps {
+  /**
+   * Container port number to be registered.
+   */
+  readonly containerPort: number,
+
+  /**
+   * Protocol of the container port.
+   *
+   * @default Protocol.TCP
+   */
+  readonly protocol?: Protocol,
+
+  /**
+   * Setting to create and attach target groups.
+   */
+  readonly targetGroups: TargetGroupProps [],
+}
+
+/**
+ * The interface for creating and attaching a target group.
+ */
+export interface TargetGroupProps {
+  /**
+   * ID for a target group to be created.
+   */
+  readonly targetGroupId: string,
+
+  /**
+   * Properties for adding new targets to a listener.
+   */
+  readonly addTargetGroupProps?: elbv2.AddApplicationTargetsProps | elbv2.AddNetworkTargetsProps,
```

Comment:
No  type unions please. If you want unions, create union-like classes instead.

Reply:
Cannot create a union for this and also listeners below because they both have the properties with the same name but different types.

Reply:
Type unions do not translate well across jsii (to all languages that we target).

Union-like classes are abstract base classes that get constructed through some factory function:

```
abstract class MyGroupUnion {
 public static variantOne(): MyGroupUnion {
    // Return some subclass of MyGroupUnion
 }

 public static variantTwo(props: SomeProps) {
    // Return some subclass of MyGroupUnion
 }
}
```

And implement